### ISSUE TITLE
Premature optimization?

### ISSUE THREAD

**Issue by Maikon** [eatoncns/ttt_core#2]
https://github.com/eatoncns/ttt_core/blob/56a3c1b4265dd34a258273f04318299d581ab8bf/lib/ttt_core/board.rb#L38

I assume this was done to avoid checking the grid for a win etc?

**Comment by eatoncns** [comment]
Yes, seemed a bit pointless doing any complicated slicing of the board when there is no possibility of a win. As this is called for every state in the negamax check it represented a reasonable percentage of the call time for the 4x4 board.

**Comment by Maikon** [comment]
Did it slow down the game significantly even after adding alpha/beta and transposition tables? Did you benchmark it?
